### ISSUE TITLE
nanogallery2 and Bootstrap 3 modals

### ISSUE THREAD

**Issue by beaver71** [nanostudio-org/nanogallery2#248]
I tried to load via javascript a gallery with several pictures (more than 80) using these options:

```
    var opts_gall = {
        itemsBaseURL: baseurl,
        thumbnailHeight: 250,
        thumbnailWidth: 250,
        thumbnailGutterHeight: 5,
        thumbnailGutterWidth: 5,
        thumbnailBorderHorizontal: 0,
        thumbnailBorderVertical: 0,
        galleryDisplayMode: "fullContent",	
        thumbnailDisplayOutsideScreen: true
    };
```

When the gallery is created on a normal div container, the entire content (all the pictures) is populated simply scrolling down in the browser window.

On the contrary when the container is inside a Bootstrap 3 modal (so div height is fixed) only a small part of pictures is shown and there is no way to add the entire content.

Here is a working fiddle to reproduce the issue:

https://jsfiddle.net/beaver71/vLmo5kc0/

**Comment by beaver71** [comment]
Obviously using `galleryDisplayMode: "pagination"` it works correctly with pagination, but this is not as good as scrolling down to see all the images.

**Comment by Kris-B** [comment]
Layout can not be evaluated when the gallery's container is not visible.
To avoid this, refresh the gallery once the modal is displayed :
`$('#your_nanogallery2').nanogallery2('refresh');`

**Comment by beaver71** [comment]
Thanks. 
I tried to refresh the gallery after modal is shown but it seems not to solve:

```
        $('#myModalGall').on('shown.bs.modal', function (e) {
          $("#gallery-all"+m).nanogallery2('refresh');
        })
        $('#myModalGall').modal('show');  
```

Fiddle updated: https://jsfiddle.net/beaver71/vLmo5kc0/13/

**Comment by Kris-B** [comment]
Fixed in v3.0.0 - https://github.com/nanostudio-org/nanogallery2/releases/tag/v3.0.0
